internal/pkg/plugin/reposcaffolding/github/golang/ishealthy.go

```diff
@@ -1,7 +1,44 @@
 package golang
 
+import (
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/merico-dev/stream/internal/pkg/log"
+	"github.com/merico-dev/stream/pkg/util/github"
+)
+
 // IsHealthy check the health for github-repo-scaffolding-golang with provided options.
 func IsHealthy(options *map[string]interface{}) (bool, error) {
-	// TODO(daniel-hutao): implement it
+	var param Param
+	if err := mapstructure.Decode(*options, &param); err != nil {
+		return false, err
+	}
+
+	if errs := validate(&param); len(errs) != 0 {
+		for _, e := range errs {
+			log.Errorf("Param error: %s", e)
+		}
+		return false, fmt.Errorf("params are illegal")
+	}
+
+	return check(&param)
```

Comment:
IsHealthy only checks the validity of the parameters, which doesn't seem enough. However, this is related to the state, and whether we shall use DTM as the single source of truth or not. It's a big decision; let's move this discussion offline.